e2e/test/scenarios/embedding/interactive-embedding.cy.spec.js

```diff
@@ -953,385 +946,245 @@ describe("scenarios > embedding > full app", () => {
+        cy.log("grant `nocollection` user access to `First collection`");
+        cy.updateCollectionGraph({
+          [ALL_USERS_GROUP]: { [FIRST_COLLECTION_ID]: "read" },
+        });
+
+        cy.signIn("nocollection");
+        startNewEmbeddingQuestion({ isMultiStageDataPicker: true });
+        selectCard({
+          cardName: cardDetails.name,
+          cardType: "model",
+          collectionNames: ["First collection"],
+        });
+        clickOnDataSource(ordersCardDetails.name);
+        verifyCardSelected({
+          cardName: cardDetails.name,
+          collectionName: "First collection",
+        });
+      });
+
+      it("should select a data source when there is no access to the immediate parent collection", () => {
+        const cardDetails = {
+          ...ordersCardDetails,
+          type: "model",
+          collection_id: THIRD_COLLECTION_ID,
+        };
+
+        cy.signInAsAdmin();
+        H.createQuestion(cardDetails);
+        cy.updateCollectionGraph({
+          [ALL_USERS_GROUP]: {
+            [FIRST_COLLECTION_ID]: "read",
+            [THIRD_COLLECTION_ID]: "read",
+          },
+        });
+
+        cy.signIn("nocollection");
+        startNewEmbeddingQuestion({ isMultiStageDataPicker: true });
+        selectCard({
+          cardName: cardDetails.name,
+          cardType: "model",
+          collectionNames: ["Third collection"],
+        });
+        clickOnDataSource(ordersCardDetails.name);
+        verifyCardSelected({
+          cardName: cardDetails.name,
+          collectionName: "Third collection",
+        });
+      });
+
+      it("should be able to join a card when the data source is a table", () => {
+        const cardDetails = {
+          ...ordersCardDetails,
+          type: "model",
+          collection_id: FIRST_COLLECTION_ID,
         };
         H.createQuestion(cardDetails);
         startNewEmbeddingQuestion({ isMultiStageDataPicker: true });
         selectTable({
-          tableName: "Orders",
+          tableName: "Products",
         });
         H.getNotebookStep("data").button("Join data").click();
         H.popover().within(() => {
           cy.icon("chevronleft").click();
           cy.icon("chevronleft").click();
-          cy.findByText("Raw Data").should("be.visible");
-          cy.findByText("Saved Questions").should("be.visible");
+        });
+        selectCard({
+          cardName: cardDetails.name,
+          cardType: "model",
+          collectionNames: ["First collection"],
+        });
+        clickOnJoinDataSource(cardDetails.name);
+        verifyCardSelected({
+          cardName: cardDetails.name,
+          collectionName: "First collection",
+        });
+      });
+    });
+
+    describe("metric", () => {
+      beforeEach(() => {
+        const cardDetails = {
+          ...ordersCountCardDetails,
+          type: "metric",
+          collection_id: null,
+        };
+        H.createQuestion(cardDetails);
+        cy.intercept({
```

Comment:
this intercept is required, so that the test would fail when we revert our code. Since both `Saved questions` and `Metrics` seem to be waiting for database endpoint before showing up, if we don't wait for the endpoint, the assertion that either saved questions or metrics don't exist would pass.